### PR TITLE
261: add unit test for public-schools-analysis model

### PR DESCRIPTION
**Pull request by trbmcginnis**
Add unit test for `public-schools-analysis` model

Add two traits `schoolsForTests` and `subdistrictTotalsTest` to the factory

### CODE REVIEW

File: frontend/mirage/factories/public-schools-analysis.js [hunk 1]
```diff
@@ -1,6 +1,86 @@
...
+          district: 17,
+          cartodb_id: 24,
+          subdistrict: 5
+        }
+      ],
+    }),
+
+
+
+  // traits ^^ //
```
Comment (julialucyhogan) [inline frontend/mirage/factories/public-schools-analysis.js]: comment & block are confusing

File: frontend/mirage/factories/public-schools-analysis.js [hunk 2]
```diff
@@ -5038,8 +5118,140 @@ export default Factory.extend({
...
+      address: "220 Henry Street",
+      grades: "6,7,8",
+      ps_enroll: 0,
+      is_enroll: 100,
+      hs_enroll: 0,
+      cartodb_id: 17,
+      source: "lcgms"
+    },
+    {
+      name: "Strawberry Sunrise",
```
Comment (julialucyhogan) [inline frontend/mirage/factories/public-schools-analysis.js]: <3 lol

File: frontend/tests/unit/models/public-schools-analysis-test.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  setupMirage(hooks);
+
+  test('calculates currentMultiplier correctly, tests that currentMultiplier is now multipliers property in futureResidentialDev', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    assert.equal(analysis.district, 17); // district is this.subDistrictFromDb[0].district
```
Comment (julialucyhogan) [inline frontend/tests/unit/models/public-schools-analysis-test.js]: district === currentMultiplier is just a random coincidence? If so, can you change the numbers so that is not the case? Makes reading the test confusing. Alternatively, create a variable called multiplier, set it to 17 and use that for the equality changes on line 17 and 19, but not this line making it explicit that the values are the same actual number but different theoretical variables

File: frontend/tests/unit/models/public-schools-analysis-test.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    const lcgmsSchools = analysis.allSchools.filter(obj => { return obj.source === 'lcgms'});
+
+    // buildings concatenates bluebook, lcgms, and scaProjects
+    assert.equal(bluebookBuildings[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsBuildings[0].name, 'Banana Bonanza')
+    assert.equal(scaProjectsBuildings[1].name, 'Donuts Delight')
+
+    // allSchools concatenates bluebook and lcgms
+    assert.equal(bluebookSchools[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsSchools[0].name, 'Banana Bonanza')
```
Comment (julialucyhogan) [inline frontend/tests/unit/models/public-schools-analysis-test.js]: while I wouldn't say we need to explicitly test that any external function "works", if we are using a function to achieve some behavior in our code (in this case, that null values are excluded from some variable?) then we should have a test that confirms that. The idea being if someone were to rewrite the code and accidentally leave out excluding of null values, then we want a test that will fail and catch that regression.

File: frontend/tests/unit/models/public-schools-analysis-test.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    // ^^ do we need to test that .compact works and that no null values are added to this allSchools concatenation?
+
+    assert.equal(analysis.lcgmsCartoIds[1], 18)
+    assert.equal(analysis.scaProjectsCartoIds[1], 52)
+    assert.equal(analysis.buildingsBldgIds[3], 'K092') // should we test that only unique values are grabbed?
```
Comment (julialucyhogan) [inline frontend/tests/unit/models/public-schools-analysis-test.js]: again, not exactly sure how this is implemented but if there is a condition or requirement of the end result of some function, test that the end result is achieved including edge cases (in this case duplicate values)

File: frontend/tests/unit/models/public-schools-analysis-test.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  test('calculates projectionOverMax correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    // maxProjection = datatables.enrollmentProjectionsMaxYear
+    assert.equal(analysis.projectionOverMax, false) // project.buildYear (2023) < maxProjection (2027), then false
+    assert.equal(analysis.buildYearMaxed, analysis.buildYear) // if projectionOverMax is true, buildYearMaxed should equal maxProjection, otherwise should equal buildYear
```
Comment (julialucyhogan) [inline frontend/tests/unit/models/public-schools-analysis-test.js]: should test the "otherwise" case as well
